pkg/labels/labels.go

```diff
@@ -200,6 +200,18 @@ func (ls Labels) Has(name string) bool {
 	return false
 }
 
+// HasDuplicateLabels returns whether ls has duplicate label names.
+func (ls Labels) HasDuplicateLabelNames() bool {
```

Comment:
We can chose here if we want more CPU (iter twice on the labels)
or more RAM (found:=make([]string,0,len(ls)))

Reply:
Labels are sorted, so you can do this in O(n) time.